### PR TITLE
fix(sync): allow local paths in vendir config

### LINKED ISSUE #189: [bug] directory source in vendir config is broken
For example, the following vendir config can't be used with myks:

```yaml
---
apiVersion: vendir.k14s.io/v1alpha1
kind: Config
directories:
  - path: ytt
    contents:
      - path: local-app
        directory:
          path: vendor/app
```

### PR DESCRIPTION
**Pull request by Zebradil**
Fixes #189

### CODE REVIEW

**Comment by Zebradil** [comment]
This is currently not working due to a vendir bug: multiple vendir processes can't run from the same directory because they use the same `.vendir-tmp` directory.

**Review by kbudde — APPROVED** [review]

**Review by kbudde — CHANGES_REQUESTED** [review]
What's the reason of the failing test?

**Comment by fritzduchardt** [comment]
> This is currently not working due to a vendir bug: multiple vendir processes can't run from the same directory because they use the same `.vendir-tmp` directory.
> 
> A solution is proposed in [carvel-dev/vendir#345](https://github.com/carvel-dev/vendir/pull/345)

Nicely done. I wonder whether this might have been the issue with struggled to reproduce when we migrated iam to myks.

File: internal/myks/sync.go [hunk 1]
```diff
@@ -140,30 +145,22 @@ func (a Application) cleanupVendorDir(vendorDir, vendirConfigFile string) error
...
-
-		relPath = relPath + string(filepath.Separator)
+		path = path + string(filepath.Separator)
 		for _, dir := range dirs {
-			log.Debug().Str("dir", dir).Str("relPath", relPath).Msg("Checking dir")
-			if dir == relPath {
-				log.Debug().Msgf("%s == %s", dir, relPath)
+			log.Debug().Str("dir", dir).Str("path", path).Msg("Checking dir")
+			if dir == path {
+				log.Debug().Msgf("%s == %s", dir, path)
```
Comment (fritzduchardt) [inline internal/myks/sync.go:152]: Can we take that one out again? For my taste syncing has become too verbose in the output log.

**Review by fritzduchardt — CHANGES_REQUESTED** [review]
Looks good and does work for me. Left a minor comment about the logging. Thanks for the quick implementation. Will start using this as soon as it is out.

Out of curiosity: did the discussed solution with the vendir directory flag not work for you factually or you just liked the ytt rewrite better?

File: internal/myks/sync.go (hunk 1 above)
Comment (Zebradil) [reply internal/myks/sync.go:152]: Yes, I'll cleanup logging there a bit. It was needed to debug the function, but when it works as expected, it is not needed.

**Comment by Zebradil** [comment]
@fritzduchardt 

> did the discussed solution with the vendir directory flag not work for you factually or you just liked the ytt rewrite better?

To be honest, I forgot about this option. But anyway, it seems to me a bit hacky. With the overlay, I don't care about what's inside the vendir config, execute `vendir sync` once and do not need additional arguments.

**Comment by kbudde** [comment]
@Zebradil if you merge/rebase dev (with #199), the tests should succeed.

**Comment by kbudde** [comment]
Ha, renovate was so "friendly" to update vendor to the latest main commit in #200.
Therefore it's not the needed branch anymore.

**Comment by Zebradil** [comment]
Strangely, I wasn't able to reproduce the issue with tests locally. But now it seems to reproduce.

**Review by kbudde — APPROVED** [review]

**Pull request merged**